services/navigation/builtin/builtin_test.go

```diff
@@ -1371,6 +1371,38 @@ func TestStartWaypoint(t *testing.T) {
 		})
 	}
 
+	t.Run("motion error returned when within planDeviation results in visiting waypoint", func(t *testing.T) {
+		s := setupStartWaypoint(ctx, t, logger)
+		defer s.closeFunc()
+
+		s.injectMS.MoveOnGlobeFunc = func(ctx context.Context, req motion.MoveOnGlobeReq) (motion.ExecutionID, error) {
+			return uuid.Nil, motion.ErrGoalWithinPlanDeviation
+		}
+
+		cancelCtx, cancelFn := context.WithTimeout(ctx, time.Millisecond*10)
```

Comment:
I'd recommend 200 ms as sometimes the CI server can be over-utilized / under provisioned & I'd recommend adding more buffer to reduce the likelihood of flakey tests